crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java

```diff
@@ -46,8 +45,8 @@ private BLS12381() {}
    * @param message The message to sign, not null
    * @return The SignatureAndPublicKey, not null
    */
-  public static SignatureAndPublicKey sign(KeyPair keyPair, byte[] message) {
-    G2Point hashInGroup2 = hashFunction(message);
+  public static SignatureAndPublicKey sign(KeyPair keyPair, byte[] message, int domain) {
```

Comment:
add the new parameter to javadoc real quick?

Reply:
It's a "domain value", but an integer? Is there a range for this value? Or is it only certain discrete values? If the former, the range should probably be documented and asserted. If it's the later, then it should probably be an enum.